arrows/core/video_input_arf.cxx

```diff
@@ -0,0 +1,558 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+// Implementation was heavily taken from
+// https://www.mathworks.com/matlabcentral/fileexchange/78652-arf_read_write
+
+#include "video_input_arf.h"
+
+#include <vital/types/image.h>
+#include <vital/types/image_container.h>
+#include <vital/types/metadata_traits.h>
+#include <vital/types/timestamp.h>
+
+#include <vital/exceptions.h>
+#include <vital/vital_types.h>
+
+#include <kwiversys/SystemTools.hxx>
+
+#include <fstream>
+#include <iostream>
+
+const uint32_t GAP_SIZE = 40;
+const uint32_t HEADER_SIZE = 104;
+const uint32_t COLOR_MAP_SIZE = 256 * 3;
+
+namespace kv = kwiver::vital;
+
+namespace kwiver {
+namespace arrows {
+namespace core {
+
+class video_input_arf::priv
+{
+public:
+  priv()
+  {
+    int num = 1;
+    if (*(char*)&num == 1)// Are we little endian?
+      convert_endian = true;// Yep, convert it
+    else
+      convert_endian = false;
+  }
+
+  // Metadata map
+  bool have_metadata_map = false;
+  vital::metadata_sptr metadata = nullptr;
+  vital::metadata_map::map_metadata_t metadata_map;
+  std::map< kv::path_t, kv::metadata_sptr > metadata_by_path;
+
+  // Utilities for little/big endian conversion
+  unsigned short byteswap(unsigned short);
+  unsigned long byteswap(unsigned long);
+  unsigned int byteswap(unsigned int);
+  unsigned __int64 byteswap(unsigned __int64);
```

Comment:
This is generating a compile error:
```
kwiver/arrows/core/video_input_arf.cxx:54:12: error: expected ‘;’ at end of member declaration
```

It looks like the compiler it interpreting this (unsurprisingly) as a member named `__int64`.

Please either use `long long`, or, since you already use `uint32_t` elsewhere, consider `uint64_t`, and `uintN_t` also for the others.